src/ngram/table.ts

```diff
@@ -1,50 +1,25 @@
-type Entry<T> = { key: T; count: number };
-export class NgramTable<T> {
-  private visited: Map<T, number>;
-  private topValue: Entry<T> = { key: null as unknown as T, count: -1 };
+type Entry<K> = { key: K; score: number };
 
-  constructor() {
-    this.visited = new Map<T, number>();
-  }
-
-  insert(key: T) {
-    const count = (this.visited.get(key) || 0) + 1;
-
-    if (this.topValue.count < count) {
-      this.topValue = { key, count };
-    }
+export class PriorityTable<K> {
+  private map: Map<K, number> = new Map<K, number>();
 
-    this.visited.set(key, count);
+  insert(key: K, score: number) {
+    const prevScore = this.map.get(key) ?? 0;
+    this.map.set(key, prevScore + score);
   }
 
-  get top(): Entry<T> | null {
-    const { key, count } = this.topValue;
+  *iter(): IterableIterator<Entry<K>> {
+    const sorted = Array.from(this.map.entries()).sort((m, n) => n[1] - m[1]);
 
-    if (count < 0) {
-      return null;
+    for (const [key, score] of sorted) {
+      yield { key, score };
     }
-
-    this.visited.delete(key);
-
-    this.topValue = { key: null as unknown as T, count: -1 };
-    for (const [key, count] of this.visited.entries()) {
-      if (count > this.topValue.count) {
-        this.topValue = { key, count };
-      }
-    }
-
-    return {
-      key,
-      count,
-    };
   }
-
-  clear() {
-    this.visited = new Map();
-    this.topValue = { key: null as unknown as T, count: -1 };
+  get size(): number {
+    return this.map.size;
   }
 
-  get size(): number {
-    return this.visited.size;
+  clear(): void {
+    this.map = new Map<K, number>();
```

Comment:
`this.map.clear()`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/clear